docs/protocol/off-chain/authentication.mdx

```diff
@@ -0,0 +1,374 @@
+---
+sidebar_position: 3
+title: Authentication
+---
+
+import Tooltip from '@site/src/components/Tooltip';
+
+# Authentication
+
+Authentication with <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">a clearnode</Tooltip> uses a secure 3-step challenge-response protocol that enables <Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps">session key</Tooltip> delegation with spending limits and operation scopes.
+
+---
+
+## Overview
+
+Authentication establishes an authenticated session between a client and <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">a clearnode</Tooltip>, enabling the client to perform private operations without repeatedly prompting the user's main wallet for signatures.
+
+### Why Session Keys?
+
+<Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps">Session keys</Tooltip> provide **flexible security management**:
+
+- **Granular Permissions**: Specify which operations the <Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps">session key</Tooltip> can perform
+- **Spending Allowances**: Set maximum spending limits per asset
+- **Time-Bounded**: Automatic expiration reduces risk of key compromise
+- **Application-Scoped**: Different keys for different apps
+- **User Experience**: No repeated wallet prompts during active session
+
+:::success Flexible Security Management
+<Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps">Session keys</Tooltip> give users a flexible way to manage security of their funds by providing specific permissions and allowances for specific apps, balancing convenience with security.
+:::
+
+### Authentication Flow
```

Comment:
Step 1:
* `auth_request` is a public endpoint, so it doesn't require a signature.
* you should mention that client should prepare parameters for this request, like: `address`, `session_key`, 'application`, `allowances` and `expires_at`; he should keep them locally at least until the Step 2.

Step 2:
* you should mention that Clearnode should validate all parameters

Step 3:
* Client should sign the EIP712 typed data using user's wallet
* currently, the response to `auth_verify` request will also contain a JWT, which can be used for subsequent connections

Reply:
[8211a936738924e9e67a3e56916ec7c6fee618be](https://github.com/layer-3/docs/pull/69/commits/8211a936738924e9e67a3e56916ec7c6fee618be)